cypress/support/fragments/data_import/match_profiles/match-profiles.js

```diff
@@ -8,17 +8,29 @@ const openNewMatchProfileForm = () => {
 };
 
 const fillExistingRecordFields = ({ field, in1, in2, subfield }) => {
-  cy.get('[name="profile.matchDetails[0].existingMatchExpression.fields[0].value"]').type(field);
-  cy.get('[name="profile.matchDetails[0].existingMatchExpression.fields[1].value"]').type(in1);
-  cy.get('[name="profile.matchDetails[0].existingMatchExpression.fields[2].value"]').type(in2);
-  cy.get('[name="profile.matchDetails[0].existingMatchExpression.fields[3].value"]').type(subfield);
+  if (!in1 && !in2 && !subfield) {
+    cy.do(TextField({ name: 'profile.matchDetails[0].existingMatchExpression.fields[0].value' }).fillIn(field));
+  } else {
+    cy.do([
+      TextField({ name: 'profile.matchDetails[0].existingMatchExpression.fields[0].value' }).fillIn(field),
+      TextField({ name: 'profile.matchDetails[0].existingMatchExpression.fields[1].value' }).fillIn(in1),
+      TextField({ name: 'profile.matchDetails[0].existingMatchExpression.fields[2].value' }).fillIn(in2),
+      TextField({ name: 'profile.matchDetails[0].existingMatchExpression.fields[3].value' }).fillIn(subfield)
+    ]);
+  }
 };
 
 const fillIncomingRecordFields = ({ field, in1, in2, subfield }) => {
-  cy.get('[name="profile.matchDetails[0].incomingMatchExpression.fields[0].value"]').type(field);
-  cy.get('[name="profile.matchDetails[0].incomingMatchExpression.fields[1].value"]').type(in1);
-  cy.get('[name="profile.matchDetails[0].incomingMatchExpression.fields[2].value"]').type(in2);
-  cy.get('[name="profile.matchDetails[0].incomingMatchExpression.fields[3].value"]').type(subfield);
+  if (!in1 && !in2 && !subfield) {
+    cy.do(TextField({ name: 'profile.matchDetails[0].incomingMatchExpression.fields[0].value' }).fillIn(field));
+  } else {
+    cy.do([
+      TextField({ name: 'profile.matchDetails[0].incomingMatchExpression.fields[0].value' }).fillIn(field),
+      TextField({ name: 'profile.matchDetails[0].incomingMatchExpression.fields[1].value' }).fillIn(in1),
+      TextField({ name: 'profile.matchDetails[0].incomingMatchExpression.fields[2].value' }).fillIn(in2),
+      TextField({ name: 'profile.matchDetails[0].incomingMatchExpression.fields[3].value' }).fillIn(subfield)
+    ]);
+  }
```

Comment:
```suggestion
  if (field) {
    cy.do(TextField({ name: 'profile.matchDetails[0].incomingMatchExpression.fields[0].value' }).fillIn(field));
  }
  if (in1) {
    cy.do(TextField({ name: 'profile.matchDetails[0].incomingMatchExpression.fields[1].value' }).fillIn(in1));
  }
  if (in2) {
    cy.do(TextField({ name: 'profile.matchDetails[0].incomingMatchExpression.fields[2].value' }).fillIn(in2));
  }
  if (subfield) {
    cy.do(TextField({ name: 'profile.matchDetails[0].incomingMatchExpression.fields[3].value' }).fillIn(subfield));
  }
```

Reply:
@OleksandrHladchenko1 , let's check every field because sometimes we have to fill only `field` or `field` and `subfield` or all the fields.

Reply:
Hi, @Javokhir12, that`s a really good point! I refactored this peace of code